File: JavaScript/7js 2/gHappy.js
/*
|--------------------------------------------------------------------------
| Simple Fun #182: Happy "g"
|--------------------------------------------------------------------------
|
*/

function gHappy(str) {
   return !str.replace(/g{2,}/g, "").includes("g");
}


function gHappy(str) {
   return !/(?<!g)g(?!g)/.test(str);
}


function gHappy(str) {
   return !/g/.test(str.replace(/g{2,}/g, ""));
}


console.log(gHappy("gg0gg3gg0gg"), true);
console.log(gHappy("gog"), false);
console.log(gHappy("ggg ggg g ggg"), false);
console.log(gHappy("A half of a half is a quarter."), true);
console.log(gHappy("good grief"), false);
console.log(gHappy("bigger is ggooder"), true);
console.log(gHappy("gggggggggg"), true);

File: Python/7py 1/g_happy.py
"""
|--------------------------------------------------------------------------
| Simple Fun #182: Happy "g"
|--------------------------------------------------------------------------
|
"""

from re import search, match, findall
def happy_g(s):
   return not search(r'(?<!g)g(?!g)', s)


def happy_g(s):
   return not match(r'(.*[^g]|^)g([^g].*|$)', s)


def happy_g(s):
   return all(len(i) > 1 for i in findall(r'g+', s))


print(happy_g("gg0gg3gg0gg"),True)
print(happy_g("gog"),False)
print(happy_g("ggg ggg g ggg"),False)
print(happy_g("A half of a half is a quarter."),True)
print(happy_g("good grief"),False)
print(happy_g("bigger is ggooder"),True)
print(happy_g("gggggggggg"),True)